### ISSUE TITLE
Fix `no-descending-specificity` false positives for different type selectors

### ISSUE THREAD

**Issue by erosman** [stylelint/stylelint#8567]
### What minimal example or steps are needed to reproduce the bug?

```css
nav {
  a {
    color: green;

    &:hover {
      background-color: #ccc;
    }
  }
}

aside {
  a {
    color: blue;

    &:hover {
      text-decoration: underline;
    }
  }
}
```

### What minimal configuration is needed to reproduce the bug?

default 

### How did you run Stylelint?

[Demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcIB2BDAbgAmAHSRjKWu++kANhAE7wYDmlMMSA3LsSQGTwAWEaMlInhL4ARijABrehACuSKAFpyVGgGIwm1sPwBfdvqSHcKAM4BLWEPyEcOjCuoZRZWTG3t8XXv0F2R+AAuMAAegYqwkJQogeYQSDTysJRk5kjunhiGeri6IAA0IABm5mQwAHIoALZwiKHVAA5lAHRgpqYF4PEltAgg-hjYIKHBCqZDNADamUOmgQCeZalI4ZBIPYpzKAoolFBD7AC6uZ1rPQBiVFUxfQBWpvGdsA0diAOzC0tpgRODIGQxGBzIb5dgfRYwZarbrmWibQLbKC7fYIP4A4LAkAnXRAA)

### Which Stylelint-related dependencies are you using?

```json
{
  "stylelint": "16.19.1",
  "stylelint-config-standard": "38.0.0"
}
```

### What did you expect to happen?

A better reporting
e.g. change error to warning 

### What actually happened?

> error Expected selector "a" to come before selector "nav a:hover" [(no-descending-specificity)](https://stylelint.io/user-guide/rules/no-descending-specificity)

**Comment by coderaiser** [comment]
I have similar case:

When running `stylelint menu/menu.css -c stylelintrc.json` with config:

```json
{
    "extends": ["stylelint-config-standard"],
    "rules": {
        "@stylistic/indentation": 4,
        "selector-class-pattern": null,
        "@stylistic/declaration-block-trailing-semicolon": "always",
        "@stylistic/declaration-colon-space-before": null,
        "@stylistic/selector-list-comma-newline-after": null,
        "comment-empty-line-before": null,
        "@stylistic/number-leading-zero": null,
        "@stylistic/number-no-trailing-zeros": null,
        "@stylistic/string-quotes": "single",
        "function-url-quotes": "never",
        "@stylistic/no-eol-whitespace": null,
        "font-family-name-quotes": "always-unless-keyword",
        "font-family-no-missing-generic-family-keyword": null,
        "rule-empty-line-before": null,
        "@stylistic/max-empty-lines": 2,
        "property-no-vendor-prefix": null,
        "selector-no-vendor-prefix": null
    },
    "plugins": [
        "@stylistic/stylelint-plugin"
    ]
}
```

on file:

```css
.menu-item-selected > .menu, .menu-item:hover > .menu  {
    display: block;
}
```


<img width="1198" height="82" alt="Image" src="https://github.com/user-attachments/assets/40f3a1e2-7253-45fb-a630-814e59bb3e21" />

Could you please help me, how should I fix my css?

**Comment by erosman** [comment]
IMHO, [no-descending-specificity](https://stylelint.io/user-guide/rules/no-descending-specificity) should not output an error when the specific rules are not known to intersect.

- Unknown intersection

```css
nav a
aside a
.abc a
```

- Known intersection

```css
html a
body a
a
nav a
aside a
```

**Comment by romainmenke** [comment]
This was resolved in 17.0.0
[Demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcIB2BDAbgAmAHSRjKWu++kANhAE7wYDmlMMSA3LsSQGTwAWEaMlInhL4ARijABrehACuSKAFpyVGgGIwm1sPwBfdvqSHcKAM4BLWEPyEcOjCuoZRZWTG3t8XXv0F2R+AAuMAAegYqwkJQogeYQSDTysJRk5kjunhiGeri6IAA0IABm5mQwAHIoALZwiKHVAA5lAHRgpqYF4PEltAgg-hjYIKHBCqZDNADamUOmgQCeZalI4ZBIPYpzKAoolFBD7AC6uZ1rPQBiVFUxfc1zizDLgZRgzQBWpvGdsA0diAOzBZLNKBCaDEBkGIwOZDfLsQEPJ7KbrmWibQLbKC7fYIcGQ4IwkAnXRAA)
